tests/kernel/usleep/src/main.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2019 Intel Corp.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <zephyr.h>
+
+/*
+ * precision timing tests in an emulation environment are not reliable.
+ * if the test passes at least once, we know it works properly, so we
+ * attempt to repeat the test RETRIES times before reporting failure.
+ */
+
+#define RETRIES		1000
+
+/*
+ * test parameters. SLEEP_US * LOOPS should be at least a few milliseconds
+ * so that the precision of the system uptime clock (1ms) isn't a factor.
+ */
+
+#define SLEEP_US	50
+#define LOOPS		200
+
+#define LOWER_BOUND_MS	((SLEEP_US * LOOPS) / 1000)
+#define UPPER_BOUND_MS	(LOWER_BOUND_MS * 2)
+
+void test_usleep(void)
+{
+	int retries = 0;
+	s64_t elapsed_ms;
+
+	while (retries < RETRIES) {
+		s64_t start_ms;
+		s64_t end_ms;
+		int i;
+
+		++retries;
+		start_ms = k_uptime_get();
+
+		for (i = 0; i < LOOPS; ++i)
```

Comment:
braces need to be added to even one liners

Reply:
@jukkar I'm happy to add braces here, but can you point out the style guideline that says they're required?  The Linux kernel guidelines specifically request that braces NOT be here, according to @ceolin they're not required by MISRA, and the Zephyr Contribution Guidelines say braces are required for `if` and `else` clauses, but says nothing about `for` that I can see.

Reply:
Hmph, I must have misinterpreted @ceolin because MISRA C:2004, 14.8 *does* require the braces.  I'll put them in.

Reply:
https://docs.zephyrproject.org/latest/contribute/index.html#coding-style

>Add braces to every if and else body, even for single-line code blocks. Use the --ignore BRACES flag to make checkpatch stop complaining.

Doesn't mention `for` loops but it's implied.